### PR TITLE
Ceds 5816 amend scenarios

### PR DESCRIPTION
**Pull request by siva1982**

### CODE REVIEW

File: src/test/resources/features/Amend Scenarios/AmendScenarios.feature [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    And I click back
+    And I navigate to declaration information page after clicking on mrn link
+    And I click Amend declaration link
+    Then I should land on Saved-Summary page
+    And I click change link for Border transport page
+    Then I should land on Border-Transport page
+    And I select Ship name as the border transport
+    And I click save and return to summary
+    Then I should land on Saved-Summary page
+    And I check the sections' headings and click confirm and continue
```
Comment (dhanist-gadhiraju) [inline src/test/resources/features/Amend Scenarios/AmendScenarios.feature:254]: is this sections ' correct

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/common/DashboardStepDef.scala [hunk 2]
```diff
@@ -19,11 +19,12 @@ package uk.gov.hmrc.test.ui.cucumber.stepdefs.common
 import uk.gov.hmrc.test.ui.cucumber.stepdefs.BaseStepDef
 import uk.gov.hmrc.test.ui.pages.common.DashboardPage
 
-class DashboardPageStepDef extends BaseStepDef {
+class DashboardStepDef extends BaseStepDef {
 
   And("""^I should land on Dashboard page""")(() => DashboardPage.checkPage())
 
   And("""^I validate declaration details on (.*) tab and check Status is (.*)""") { (tab: String, status: String) =>
+    DashboardPage.refreshPage()
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/common/DashboardStepDef.scala:27]: why do we need refresh here when you already have status refresh inside this validate method

File: src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala [hunk 3]
```diff
@@ -41,6 +41,10 @@ trait DriverHelper {
 
   def continue(): Unit = clickById("submit")
 
+  def saveAndReturnToSummary(): Unit = clickById("save_and_return_to_summary")
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala:44]: Should this method be in driver helper also below one

File: src/test/scala/uk/gov/hmrc/test/ui/pages/common/DashboardPage.scala [hunk 4]
```diff
@@ -62,6 +62,10 @@ object DashboardPage extends BasePage {
     findChildByClassName(mrnCell, "govuk-link")
   }
 
+  def refreshPage(): Unit = {
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/common/DashboardPage.scala:65]: you can remove this

File: src/test/scala/uk/gov/hmrc/test/ui/pages/common/DeclarationInformationPage.scala [hunk 5]
```diff
@@ -95,13 +96,47 @@ object DeclarationInformationPage extends BasePage {
 
     // store url for checking back navigation on cancel and copy declaration
     val url = driver.getCurrentUrl
-    store(DeclarationInfoPath -> Detail(driver.getCurrentUrl.substring(url.indexOf("/submissions"))))
+    val declarationInfoPath = Detail(driver.getCurrentUrl.substring(url.indexOf("/submissions")))
+    storeOne(DeclarationInfoPath -> declarationInfoPath)
+   // cacheForAmendments.put(DeclarationInfoPath, declarationInfoPath)
```
Comment (dhanist-gadhiraju) [inline src/test/scala/uk/gov/hmrc/test/ui/pages/common/DeclarationInformationPage.scala:101]: any reason why this is commented out.

File: run_tag.sh [hunk 6]
```diff
@@ -19,4 +19,5 @@ sbt clean \
    -Dbrowser=$Browser \
    -Denv=$Environment \
    -Dcucumber.filter.tags=$Tag \
+   -Dbrowser.option.headless=false
```
[the remark this answers is not in the document]
Comment (siva1982) [reply run_tag.sh:22]: We need this

File: src/test/resources/features/Amend Scenarios/AmendScenarios.feature (hunk 1 above)
Comment (siva1982) [reply src/test/resources/features/Amend Scenarios/AmendScenarios.feature:254]: yes it is

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/common/DashboardStepDef.scala (hunk 2 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/common/DashboardStepDef.scala:27]: No, while validating the details, first time it is not showing the actual result in the dashboard page, so when we refresh the page it is getting the updated status

File: src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala (hunk 3 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala:44]: it is in the driver helper class

File: src/test/scala/uk/gov/hmrc/test/ui/pages/common/DashboardPage.scala (hunk 4 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/common/DashboardPage.scala:65]: This is required to refresh the page after we do amend

File: src/test/scala/uk/gov/hmrc/test/ui/pages/common/DeclarationInformationPage.scala (hunk 5 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/common/DeclarationInformationPage.scala:101]: removed

File: src/test/scala/uk/gov/hmrc/test/ui/pages/common/DetailKeys.scala [hunk 7]
```diff
@@ -23,7 +23,7 @@ object DetailKeys {
   val sectionCommon = 7
 
   val MrnOnDashboard: DetailKey = DetailKey("Mrn on Dashboard", sectionCommon, skipRowCheck = true)
-  val StatusOnDashboard: DetailKey = DetailKey("Status on Dashboard", sectionCommon, skipRowCheck = true)
+  var StatusOnDashboard: DetailKey = DetailKey("Status on Dashboard", sectionCommon, skipRowCheck = true)
```
[the remark this answers is not in the document]
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/common/DetailKeys.scala:26]: done

File: src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/common/DashboardStepDef.scala (hunk 2 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/cucumber/stepdefs/common/DashboardStepDef.scala:27]: removed now

File: src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala (hunk 3 above)
Comment (siva1982) [reply src/test/scala/uk/gov/hmrc/test/ui/pages/base/DriverHelper.scala:44]: Done

**Review by dhanist-gadhiraju — APPROVED** [review]

**Review by dhanist-gadhiraju — DISMISSED** [review]

**Review by dhanist-gadhiraju — DISMISSED** [review]

**Review by magicaljellybeans — APPROVED** [review]

**Pull request merged**